api/v1beta1/nutanixmachine_types.go

```diff
@@ -47,7 +47,30 @@ const (
 	NutanixMachineBootstrapRefKindImage = "Image"
 )
 
+// NutanixImageLookup defines how to fetch images for the cluster
+// using the fields combined.
+type NutanixImageLookup struct {
+	// Format is the naming format to look up the image for this
+	// machine It will be ignored if an explicit image is set. Supports
+	// substitutions for {{.BaseOS}} and {{.K8sVersion}} with the base OS and
+	// kubernetes version, respectively. The BaseOS will be the value in
+	// BaseOS and the K8sVersion is the value in the Machine .spec.version, with the v prefix removed.
+	// This is effectively the defined by the packages produced by kubernetes/release without v as a
+	// prefix: 1.13.0, 1.12.5-mybuild.1, or 1.17.3. For example, the default
+	// image format of {{.BaseOS}}-?{{.K8sVersion}}-* and BaseOS as "rhel-8.10" will end up
+	// searching for images that match the pattern rhel-8.10-1.30.5-* for a
+	// Machine that is targeting kubernetes v1.30.5. See
+	// also: https://golang.org/pkg/text/template/
+	// +kubebuilder:default:="capx-{{.BaseOS}}-{{.K8sVersion}}-*"
+	Format *string `json:"format,omitempty"`
+	// BaseOS is the name of the base operating system to use for
+	// image lookup.
+	// +kubebuilder:validation:MinLength:=1
+	BaseOS string `json:"baseOS"`
+}
+
 // NutanixMachineSpec defines the desired state of NutanixMachine
+// +kubebuilder:validation:XValidation:rule="has(self.image) != has(self.imageLookup)",message="Either 'image' or 'imageLookup' must be set, but not both"
```

Comment:
This CEL expression is pretty opaque but it does work!

Reply:
This is tested in main_test.go with several scenarios.